tests/backup/backup_nfs_test.go

```diff
@@ -0,0 +1,265 @@
+		for i := 0; i < 5; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				appNamespaces = append(appNamespaces, namespace)
+				scheduledAppContexts = append(scheduledAppContexts, ctx)
+			}
+		}
+		log.Infof("The list of namespaces deployed are", appNamespaces)
+
+	})
+	It("To validate backup and restore status after deleting the respective nfs executor pod while in progress", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step("Creating NFS backup location", func() {
+			log.InfoD("Creating NFS backup location")
+			backupLocationProviders := getProviders()
+			for _, provider := range backupLocationProviders {
+				bkpLocationName = fmt.Sprintf("%s-%s-%s", provider, getGlobalBucketName(provider), RandomString(10))
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				err := CreateBackupLocation(provider, bkpLocationName, backupLocationUID, "", "", getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating NFS backup location %s", bkpLocationName))
+			}
+		})
+
+		Step("Registering application clusters for backup", func() {
+			log.InfoD("Registering application clusters for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateApplicationClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			log.FailOnError(err, "Fetching [%s] uid", SourceClusterName)
+		})
+
+		Step("Create schedule policy", func() {
+			log.InfoD("Creating schedule policy")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			schedulePolicyName = fmt.Sprintf("%s-%v", "periodic-schedule-policy", RandomString(5))
+			schedulePolicyInfo = Inst().Backup.CreateIntervalSchedulePolicy(5, schedulePolicyInterval, 5)
+			userSchedulePolicyCreateRequest := &api.SchedulePolicyCreateRequest{
+				CreateMetadata: &api.CreateMetadata{
+					Name:  schedulePolicyName,
+					OrgId: orgID,
+				},
+				SchedulePolicy: schedulePolicyInfo,
+			}
+			_, err = Inst().Backup.CreateSchedulePolicy(ctx, userSchedulePolicyCreateRequest)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation schedule policy %s", schedulePolicyName))
+			schedulePolicyUID, err = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, schedulePolicyName)
+			log.FailOnError(err, "failed to fetch uid for schedule policy- %s", schedulePolicyName)
+		})
+
+		Step("Taking schedule backup of multiple namespaces", func() {
```

Comment:
Can we move this entire thing into a loop ? Like have a dict and say run for 3 times for all three variations?

Reply:
Not sure why we need a loop and switch case here? We can simply split this into 3 different steps right?
One step each for schedule backup, custom resource backup and multi namespace backup.

Reply:
@kshithijiyer-px @mkoppal-px 
Have made the changes as suggested by @kshithijiyer-px and I also feel that the code is more clean and less redundant now.
Initially I had 3 function repeated each for schedule backup, custom resource backup and multi namespace backup.